### DISCUSSION TITLE
How to add context menu to user's right click?

### DISCUSSION THREAD

**Discussion by billywatkins** [tauri-apps/tauri#8726] [General]
Hey,

I am trying to add a menu option to the user's right-click menu. I can't find any docs on this. Is it possible to add an option to the user's right-click? For example, when they select a file, our option should show up, like "Change the wallpaper."

for ex:
<img width="790" alt="Vivid blurred colorful background context menu" src="https://github.com/tauri-apps/tauri/assets/124872003/c9e857dd-a4cd-4039-9c17-eea8f1d970f3">

**Comment by billywatkins** [comment]
I found this in MUDA  https://github.com/tauri-apps/muda?tab=readme-ov-file#context-menus-popup-menus. Is there any documentation on how to use this?

Thank you in advance.

**Comment by ok-nick** [comment]
Hi, I implemented this for my app on macOS (currently). You can check out the project [here](https://github.com/ok-nick/c2pa-preview) and the finder extension code [here](https://github.com/ok-nick/c2pa-preview/blob/main/backend/src/file_ext/finder.rs) and [here](https://github.com/ok-nick/c2pa-preview/blob/main/backend/Info.plist).

I decided not to implement a "[finder sync extension](https://developer.apple.com/documentation/findersync)" because they require compiling a separate `.appex` program and you'd have to use IPC which can get cumbersome. Rather, I used (the less powerful) [NSServices](https://developer.apple.com/documentation/bundleresources/information_property_list/nsservices) which you can find defined in my [Info.plist](https://github.com/ok-nick/c2pa-preview/blob/main/backend/Info.plist) file.

It looks something like this:
<img width="1031" alt="finder" src="https://github.com/tauri-apps/tauri/assets/25470747/6639ebd9-d24a-4bfa-9217-3d526abdf42f">

**Reply by iHTCboy** [reply]
@billywatkins @ok-nick  If it is a Windows system, do you have any ideas or methods to implement this menu? Thank you~
